### ISSUE TITLE
Wrong working templates with specific theme name

### ISSUE THREAD

**Issue by quartze** [woothemes/woocommerce#33591]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

If we name our theme with a name containing `-` (e.g. main-theme), then woocommerce does not read all templates from the / templates folder. This behaves randomly and only reads certain files (usually from the main folder).

### Expected behavior

Reading full template from `woocommerce` folder.

### Actual behavior

Some of files (e.g. /cart/cart.php) does not work even if we delete context in files.

### Steps to reproduce

1. Create a theme called `main-theme`
2. Copy the /templates folder from woocommerce plugin then put it into your theme
3. Try change something and check if it's changed on site (e.g. /cart/cart.php)

### WordPress Environment

```
### WordPress Environment ###

WordPress address (URL): https://woocommerce-542486-2665533.cloudwaysapps.com
Site address (URL): https://woocommerce-542486-2665533.cloudwaysapps.com
WC Version: 6.6.1
REST API Version: ✔ 6.6.1
WC Blocks Version: ✔ 7.6.2
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: pl_PL
External object cache: ✔

### Server Environment ###

Server Info: Apache/2.4.25 (Debian)
PHP Version: 7.4.29
PHP Post Max Size: 64 MB
PHP Time Limit: 300
PHP Max Input Vars: 5000
cURL Version: 7.52.1
OpenSSL/1.0.2u

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.1.48-MariaDB-1~stretch
Max Upload Size: 64 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.6.1
WC Database Prefix: wp_
Całkowity rozmiar bazy danych: 3.13MB
Rozmiar bazy danych: 1.44MB
Rozmiar indeksu bazy danych: 1.69MB
wp_woocommerce_sessions: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_woocommerce_api_keys: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_woocommerce_attribute_taxonomies: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_woocommerce_downloadable_product_permissions: Dane: 0.02MB + Indeks: 0.06MB + Silnik InnoDB
wp_woocommerce_order_items: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_woocommerce_order_itemmeta: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_woocommerce_tax_rates: Dane: 0.02MB + Indeks: 0.06MB + Silnik InnoDB
wp_woocommerce_tax_rate_locations: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_woocommerce_shipping_zones: Dane: 0.02MB + Indeks: 0.00MB + Silnik InnoDB
wp_woocommerce_shipping_zone_locations: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_woocommerce_shipping_zone_methods: Dane: 0.02MB + Indeks: 0.00MB + Silnik InnoDB
wp_woocommerce_payment_tokens: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_woocommerce_payment_tokenmeta: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_woocommerce_log: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_actionscheduler_actions: Dane: 0.02MB + Indeks: 0.11MB + Silnik InnoDB
wp_actionscheduler_claims: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_actionscheduler_groups: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_actionscheduler_logs: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_bv_fw_requests: Dane: 0.02MB + Indeks: 0.00MB + Silnik InnoDB
wp_bv_ip_store: Dane: 0.19MB + Indeks: 0.14MB + Silnik InnoDB
wp_bv_lp_requests: Dane: 0.02MB + Indeks: 0.00MB + Silnik InnoDB
wp_commentmeta: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_comments: Dane: 0.02MB + Indeks: 0.09MB + Silnik InnoDB
wp_links: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_options: Dane: 0.13MB + Indeks: 0.03MB + Silnik InnoDB
wp_postmeta: Dane: 0.06MB + Indeks: 0.03MB + Silnik InnoDB
wp_posts: Dane: 0.09MB + Indeks: 0.06MB + Silnik InnoDB
wp_termmeta: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_terms: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_term_relationships: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_term_taxonomy: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_usermeta: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_users: Dane: 0.02MB + Indeks: 0.05MB + Silnik InnoDB
wp_wc_admin_notes: Dane: 0.05MB + Indeks: 0.00MB + Silnik InnoDB
wp_wc_admin_note_actions: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_wc_category_lookup: Dane: 0.02MB + Indeks: 0.00MB + Silnik InnoDB
wp_wc_customer_lookup: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_wc_download_log: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_wc_order_coupon_lookup: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_wc_order_product_lookup: Dane: 0.02MB + Indeks: 0.06MB + Silnik InnoDB
wp_wc_order_stats: Dane: 0.02MB + Indeks: 0.05MB + Silnik InnoDB
wp_wc_order_tax_lookup: Dane: 0.02MB + Indeks: 0.03MB + Silnik InnoDB
wp_wc_product_attributes_lookup: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_wc_product_download_directories: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_wc_product_meta_lookup: Dane: 0.02MB + Indeks: 0.09MB + Silnik InnoDB
wp_wc_rate_limits: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_wc_reserved_stock: Dane: 0.02MB + Indeks: 0.00MB + Silnik InnoDB
wp_wc_tax_rate_classes: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_wc_webhooks: Dane: 0.02MB + Indeks: 0.02MB + Silnik InnoDB
wp_wpr_rucss_resources: Dane: 0.02MB + Indeks: 0.06MB + Silnik InnoDB
wp_wpr_rucss_used_css: Dane: 0.02MB + Indeks: 0.05MB + Silnik InnoDB

### Post Type Counts ###

acf-field: 8
acf-field-group: 1
attachment: 3
customize_changeset: 1
nav_menu_item: 17
page: 13
post: 8
product: 7
revision: 20
wp_global_styles: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Komunikaty o błędach nie powinny być widoczne dla użytkowników.

### Active Plugins (8) ###

Advanced Custom Fields PRO: przez Delicious Brains – 5.12.2
Klaviyo: przez Klaviyo
Inc. – 3.0.3

Loco Translate: przez Tim Whitlock – 2.6.2
MalCare Security - Free Malware Scanner, Protection & Security for WordPress: przez MalCare Security – 4.67
Show Current Template: przez JOTAKI Taisuke – 0.4.6
Woocommerce Side Cart Premium: przez XootiX – 3.1
WooCommerce: przez Automattic – 6.6.1
WP Rocket: przez WP Media – 3.10.7

### Inactive Plugins (3) ###

Advanced Custom Fields: przez Delicious Brains – 5.12.2
Ninja Forms: przez Saturday Drive – 3.6.11
Yoast SEO: przez Team Yoast – 19.0

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Object Cache Pro (Drop-in)

### Must Use Plugins (1) ###

Object Cache Pro (MU): przez Rhubarb Group – 1.14.5

### Settings ###

API Enabled: –
Force SSL: –
Currency: PLN (zł)
Currency Position: right_space
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Strona sklepu: #6 - /sklep
Koszyk: #7 - /cart
Zamówienie: #8 - /checkout
Moje konto: #9 - /my-account
Regulamin: ❌ Strona nie ustawiona

### Theme ###

Name: Kwiatowa dostawa
Version: 1.0
Author URL: 
Child Theme: ❌

WooCommerce Support: ✔

### Templates ###

Overrides: kwiatowadostawa/woocommerce/archive-product.php
kwiatowadostawa/woocommerce/cart/cart.php
kwiatowadostawa/woocommerce/content-product.php wersja 3.4.0 jest nieaktualna. Bazowa wersja to 3.6.0
kwiatowadostawa/woocommerce/content-single-product.php
kwiatowadostawa/woocommerce/single-product/price.php
kwiatowadostawa/woocommerce/single-product-reviews.php
kwiatowadostawa/woocommerce/single-product.php

Outdated Templates: ❌
					
					
						Dowiedz się jak zaktualizować

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
wc-pay-promotion
wc-pay-welcome-page
wc-pay-subscriptions-page

Disabled Features: minified-js
settings

Daily Cron: ✔ Next scheduled: 2022-06-24 12:11:07 +00:00
Options: ✔
Notes: 47
Onboarding: completed

### Action Scheduler ###

Zakończ: 32
Oldest: 2022-06-03 13:28:14 +0000
Newest: 2022-06-24 06:12:15 +0000

Oczekujące: 1
Oldest: 2022-06-25 06:12:15 +0000
Newest: 2022-06-25 06:12:15 +0000


### Status report information ###

Generated at: 2022-06-24 07:04:33 +00:00
```

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Comment by woocommercebot** [comment]
We are adding the `status: needs reproduction` label to this issue to try reproduce it on the   current released version of WooCommerce.

  Thank you for your patience.

**Comment by jorgeatorres** [comment]
Hi @quartze!

I apologize for the delay here. Is this still a problem on your site?
We tried to reproduce it but couldn't. Even using your example theme name (`main-theme`) we were able to override the templates.

One thing I did notice is that you're running your site with an object cache. That shouldn't be a problem, but template file paths are cached, so I was wondering if it could have anything to do with the problems you were seeing. Have you tried testing with the cache disabled?

Looking forward to hearing back from you.
